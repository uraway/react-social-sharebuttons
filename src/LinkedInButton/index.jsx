// @flow
import React, { Component } from 'react';

type Props = {
  url?: string,
  counter?: 'top' | 'right',
  lang?: string,
};

export default class LinkedinButton extends Component<Props> {
  node = null;

  static defaultProps = {
    url: typeof window !== 'undefined' ? window.url : '',
    counter: '',
    lang: 'ja',
  };

  componentDidMount() {
    const { lang } = this.props;

    if (window.IN) delete window.IN;
    const s = document.createElement('script');
    s.src = '//platform.linkedin.com/in.js';
    s.type = 'text/javascript';
    if (lang) s.lang = lang;
    if (this.node && this.node.parentNode) this.node.parentNode.appendChild(s);
  }

  render() {
    const { url, counter } = this.props;
    return (
      <div>
        <script ref={(node) => (this.node = node)} type="IN/Share" data-url={url} data-counter={counter} />
      </div>
    );
  }
}
