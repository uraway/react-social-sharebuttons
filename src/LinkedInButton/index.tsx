import { FC, useEffect } from 'react';

export type LinkedInButtonProps = {
  url?: string;
  counter?: 'top' | 'right';
  lang?: string;
};

declare let window: Window & { IN: unknown };

const LinkedInButton: FC<LinkedInButtonProps> = (props) => {
  const { lang, url, counter } = props;

  useEffect(() => {
    if (window.IN) delete window.IN;
    const script = document.createElement('script');
    script.src = 'https://platform.linkedin.com/in.js';
    script.type = 'text/javascript';
    if (lang) script.lang = lang;
    document.getElementsByTagName('head')[0]?.appendChild(script);

    return () => script?.remove();
  }, [lang, props]);

  return <script type="IN/Share" data-url={url} data-counter={counter} />;
};

LinkedInButton.defaultProps = {
  url: typeof window !== 'undefined' ? window.location.href : '',
  counter: 'right',
  lang: 'ja',
};

export default LinkedInButton;
