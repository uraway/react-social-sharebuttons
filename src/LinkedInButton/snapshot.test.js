import React from 'react';
import renderer from 'react-test-renderer';
import LinkedInButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<LinkedInButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer.create(<LinkedInButton url="url" counter="right" lang="ja" />).toJSON();
  expect(tree).toMatchSnapshot();
});
