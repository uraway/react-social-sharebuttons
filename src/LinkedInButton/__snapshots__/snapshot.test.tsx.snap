// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<script
  data-counter="right"
  data-url="url"
  type="IN/Share"
/>
`;

exports[`renders correctly with required props 1`] = `
<script
  data-counter="right"
  data-url="http://localhost/"
  type="IN/Share"
/>
`;
